<p>Each tag should be documented with a short statement explaining its intent.</p>
<p>Tags must start with capital letter</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<h3>JSON</h3>
<pre>
{
    "swagger": "2.0", 
    "info": {
        "version": "1.0.0", 
        "title": "Swagger Petstore"
    }, 
    "tags": [
        {
            "name": "unused-tag"
        }, 
        {
            "name": "some-tag"
        }
    ], 
    "paths": {
        "/pets": {
            "get": {
                "tags": [
                    "some-tag"
                ], 
                "responses": {
                    "200": {
                        "description": "OK"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: unused-tag
- name: some-tag
paths:
  /pets:
    get:
      tags:
      - some-tag
      responses:
        200:
          description: OK
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<h3>JSON</h3>
<pre>
{
    "swaggeer": "2.0", 
    "info": {
        "version": "1.0.0", 
        "title": "Swagger Petstore"
    }, 
    "tags": [
        {
            "name": "Some-tag", 
            "description": "a tag referenced in the operations"
        }, 
        {
            "name": "Other-tag", 
            "description": "another tag referenced in the operations"
        }
    ], 
    "paths": {
        "/pets": {
            "get": {
                "tags": [
                    "Some-tag", 
                    "Other-tag"
                ], 
                "responses": {
                    "200": {
                        "description": "OK"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
swaggeer: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: Some-tag
  description: a tag referenced in the operations
- name: Other-tag
  description: another tag referenced in the operations
paths:
  /pets:
    get:
      tags:
      - Some-tag
      - Other-tag
      responses:
        200:
          description: OK
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<h3>JSON</h3>
<pre>
{
    "openapi": "3.0.1", 
    "info": {
        "version": "1.0.0", 
        "title": "Swagger Petstore"
    }, 
    "tags": [
        {
            "name": "unused-tag"
        }, 
        {
            "name": "some-tag"
        }
    ], 
    "paths": {
        "/pets": {
            "get": {
                "tags": [
                    "some-tag"
                ], 
                "responses": {
                    "200": {
                        "description": "OK"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
openapi: "3.0.1"
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: unused-tag
- name: some-tag
paths:
  /pets:
    get:
      tags:
      - some-tag
      responses:
        200:
          description: OK
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<h3>JSON</h3>
<pre>
{
    "openapi": "3.0.1", 
    "info": {
        "version": "1.0.0", 
        "title": "Swagger Petstore"
    }, 
    "tags": [
        {
            "name": "Some-tag", 
            "description": "a tag referenced in the operations"
        }, 
        {
            "name": "Other-tag", 
            "description": "another tag referenced in the operations"
        }
    ], 
    "paths": {
        "/pets": {
            "get": {
                "tags": [
                    "Some-tag", 
                    "Other-tag"
                ], 
                "responses": {
                    "200": {
                        "description": "OK"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
openapi: "3.0.1"
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: Some-tag
  description: a tag referenced in the operations
- name: Other-tag
  description: another tag referenced in the operations
paths:
  /pets:
    get:
      tags:
      - Some-tag
      - Other-tag
      responses:
        200:
          description: OK
</pre>
