<p>Responses with code 204 should not return content.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<h3>JSON</h3>
<pre>
{
    "swagger": "2.0", 
    "info": {
        "version": "1.0.0", 
        "title": "Swagger Petstore"
    }, 
    "paths": {
        "/pets/{petId}": {
            "delete": {
                "parameters": [
                    {
                        "in": "path", 
                        "name": "petId", 
                        "required": true, 
                        "type": "string", 
                        "format": "uuid"
                    }
                ], 
                "responses": {
                    "204": {
                        "description": "Deleted", 
                        "schema": {
                            "type": "object"
                        }
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets/{petId}:
    delete:
      parameters:
        - in: path
          name: petId
          required: true
          type: string
          format: uuid
      responses:
        204:
          description: Deleted
          schema:
            type: object
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<h3>JSON</h3>
<pre>
{
    "swagger": "2.0", 
    "info": {
        "version": "1.0.0", 
        "title": "Swagger Petstore"
    }, 
    "paths": {
        "/pets/{petId}": {
            "delete": {
                "parameters": [
                    {
                        "in": "path", 
                        "name": "petId", 
                        "required": true, 
                        "type": "string", 
                        "format": "uuid"
                    }
                ], 
                "responses": {
                    "204": {
                        "description": "Deleted"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets/{petId}:
    delete:
      parameters:
        - in: path
          name: petId
          required: true
          type: string
          format: uuid
      responses:
        204:
          description: Deleted
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<h3>JSON</h3>
<pre>
{
    "openapi": "3.0.1", 
    "info": {
        "version": "1.0.0", 
        "title": "Swagger Petstore"
    }, 
    "paths": {
        "/pets/{petId}": {
            "delete": {
                "parameters": [
                    {
                        "in": "path", 
                        "name": "petId", 
                        "required": true, 
                        "schema": {
                            "type": "string", 
                            "format": "uuid"
                        }
                    }
                ], 
                "responses": {
                    "204": {
                        "description": "Deleted", 
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object"
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
openapi: 3.0.1
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets/{petId}:
    delete:
      parameters:
        - in: path
          name: petId
          required: true
          schema:
            type: string
            format: uuid
      responses:
        204:
          description: Deleted
          content:
            application/json:
              schema:
                type: object
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<h3>JSON</h3>
<pre>
{
    "openapi": "3.0.1", 
    "info": {
        "version": "1.0.0", 
        "title": "Swagger Petstore"
    }, 
    "paths": {
        "/pets/{petId}": {
            "delete": {
                "parameters": [
                    {
                        "in": "path", 
                        "name": "petId", 
                        "required": true, 
                        "schema": {
                            "type": "string", 
                            "format": "uuid"
                        }
                    }
                ], 
                "responses": {
                    "204": {
                        "description": "Deleted"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
openapi: 3.0.1
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets/{petId}:
    delete:
      parameters:
        - in: path
          name: petId
          required: true
          schema:
            type: string
            format: uuid
      responses:
        204:
          description: Deleted
</pre>