<p>The file with the OpenAPI specification cannot contain empty lines, this will prevent its analysis.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info: 
  title: Ejemplo de API
  version: 1.0.0
paths:

  /clientes:
  
    get:
      # 'summary': Es OBLIGATORIO usar este elemento en todos los paths. Se trata de una breve descripción del endpoint
      summary: Obtención de una colección de clientes
      # 'description': es OBLIGATORIO que el contenido de este campo esté cumplimentado y difiera del del campo 'summary'
      # Debe darse una descripción lo más detallada posible de la funcionalidad del endpoint sin describir la lógica de negocio.
      # Si el endpoint está protegido por un scope Oauth debe indicarse cuál és en la descripción.
      description: |
        Permite obtener una colección de clientes.
        Por ejemplo: /usuarios o /expedientes.
        Puede admitir cabeceras, *query parameters* y el objeto $filter en el body.

        **Scope Oauth:** clientes_sc_consulta
      
      responses:
        200:
          description: OK
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info: 
  title: Ejemplo de API
  version: 1.0.0
paths:
  /clientes:
    get:
      # 'summary': Es OBLIGATORIO usar este elemento en todos los paths. Se trata de una breve descripción del endpoint
      summary: Obtención de una colección de clientes
      # 'description': es OBLIGATORIO que el contenido de este campo esté cumplimentado y difiera del del campo 'summary'
      # Debe darse una descripción lo más detallada posible de la funcionalidad del endpoint sin describir la lógica de negocio.
      # Si el endpoint está protegido por un scope Oauth debe indicarse cuál és en la descripción.
      description: |
        Permite obtener una colección de clientes.
        Por ejemplo: /usuarios o /expedientes.
        Puede admitir cabeceras, *query parameters* y el objeto $filter en el body.
        **Scope Oauth:** clientes_sc_consulta
      responses:
        200:
          description: OK
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1

info:
  title: Ejemplo de API
  version: 1.0.0

paths:

  /clientes:

    get:
      summary: Obtención de una colección de clientes
      description: |
        Permite obtener una colección de clientes.
        Por ejemplo: /usuarios o /expedientes.
        Puede admitir cabeceras, *query parameters* y el objeto $filter en el body.

        **Scope Oauth:** clientes_sc_consulta
      responses:
        200:
          description: OK
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Ejemplo de API
  version: 1.0.0
paths:
  /clientes:
    get:
      summary: Obtención de una colección de clientes
      description: |
        Permite obtener una colección de clientes.
        Por ejemplo: /usuarios o /expedientes.
        Puede admitir cabeceras, *query parameters* y el objeto $filter en el body.
        **Scope Oauth:** clientes_sc_consulta
      responses:
        200:
          description: OK
</pre>