<p>There can be at most one body parameter in each operation. This violation applies
to OpenAPI 2 only.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post:
      parameters:
      - in: body
        name: someParam
        required: true
        schema:
          type: string
      - in: body
        name: otherParam
        required: true
        schema:
          type: string
      responses:
        default:
          description: the default response
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post:
      parameters:
      - in: body
        name: someParam
        required: true
        schema:
          type: string
      - in: query
        name: otherParam
        required: true
        schema:
          type: string
      responses:
        default:
          description: the default response
</pre>
