<p>The defined mime types (for specification v2) or media types (for specification v3) must conform to the <a href="https://tools.ietf.org/html/rfc6838">RFC6838</a> and <a href="https://tools.ietf.org/html/rfc7231">RFC7231</a>.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
</pre>