<p>El par치metro elegido debe definirse en esta operaci칩n. Por defecto, $expand</p>
<h2>Ejemplo de c칩digo no compatible (OpenAPI 2)</h2>
<h3>Todos los ejemplos est치n en YAML</h3>
<pre>
  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters: 
          - in: query
            name: other 
            type: array
            items:
              type: string
          - in: query
            name: hola 
            type: array
            items:
              type: string
        responses:
          206:
            description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $expand
            type: array
            items:
              type: string
          - in: query
            name: $expand
            type: array
            items:
              type: string
        responses:
          206:
            description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters: 
          - in: query
            name: other 
            schema:
              type: array
              items:
                type: string
          - in: query
            name: select 
            schema:
              type: array
              items:
                type: string
        responses:
          206:
            description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $expand
            schema:
              type: array
              items:
                type: string
          - in: query
            name: $expand
            schema:
              type: array
              items:
                type: string
        responses:
          206:
            description: Ok
</pre>