<p>Define the "default" response is mandatory.</p>
<p>The "default" response is used for all HTTP codes that are not covered individually.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<h3>All the examples are in YAML</h3>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    get:
      responses:
        200:
          description: OK
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    get:
      responses:
        200:
          description: OK
        default:
          description: Error
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    get:
      responses:
        200:
          description: OK
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    get:
      responses:
        200:
          description: OK
        default:
          description: Error
</pre>