<p>Depending of the verb, some path structures are not encouraged. This can be a hint of a design flaw.</p>
<h2>Allowed paths</h2>
<table>
  <thead>
    <tr>
      <th>VERBO</th>
      <th>ESTRUCTURA</th>
    </tr>
  </thead>
  <tbody>
  <tr>
    <td>GET</td>
    <td>/resources</td>
  </tr>
  <tr>
    <td>GET</td>
    <td>/resources/{id}</td>
  </tr>
  <tr>
    <td>GET</td>
    <td>/resources/{id}/subresources</td>
  </tr>
  <tr>
    <td>POST</td>
    <td>/resources</td>
  </tr>
  <tr>
    <td>POST</td>
    <td>/resources/get</td>
  </tr>
  <tr>
    <td>POST</td>
    <td>/resources/delete</td>
  </tr>
  <tr>
    <td>POST</td>
    <td>/resources/{id}/subresources</td>
  </tr>
  <tr>
    <td>PUT</td>
    <td>/resources/{id}</td>
  </tr>
  <tr>
    <td>PATCH</td>
    <td>/resources/{id}</td>
  </tr>
  <tr>
    <td>DELETE</td>
    <td>/resources/{id}</td>
  </tr>
  </tbody>
</table>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<h3>JSON</h3>
<pre>
{
    "swagger": "2.0", 
    "info": {
        "title": "Swagger Petstore", 
        "version": "1.0.0"
    }, 
    "paths": {
        "/customers/{customerId}": {
            "post": {
                "parameters": [
                    {
                        "name": "customerId", 
                        "in": "path", 
                        "required": true, 
                        "type": "integer"
                    }, 
                    {
                        "name": "body", 
                        "in": "body", 
                        "required": true, 
                        "schema": {
                            "type": "object"
                        }
                    }
                ], 
                "responses": {
                    "200": {
                        "description": "Ok"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /customers/{customerId}:
    post:
      parameters:
      - name: customerId
        in: path
        required: true
        type: integer
      - name: body
        in: body
        required: true
        schema:
          type: object
      responses:
        200:
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<h3>JSON</h3>
<pre>
{
    "swagger": "2.0", 
    "info": {
        "title": "Swagger Petstore", 
        "version": "1.0.0"
    }, 
    "paths": {
        "/customers/{customerId}": {
            "put": {
                "parameters": [
                    {
                        "name": "customerId", 
                        "in": "path", 
                        "required": true, 
                        "type": "integer"
                    }, 
                    {
                        "name": "body", 
                        "in": "body", 
                        "required": true, 
                        "schema": {
                            "type": "object"
                        }
                    }
                ], 
                "responses": {
                    "200": {
                        "description": "Ok"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /customers/{customerId}:
    put:
      parameters:
      - name: customerId
        in: path
        required: true
        type: integer
      - name: body
        in: body
        required: true
        schema:
          type: object
      responses:
        200:
          description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<h3>JSON</h3>
<pre>
{
    "openapi": "3.0.1", 
    "info": {
        "title": "Swagger Petstore", 
        "version": "1.0.0"
    }, 
    "servers": [
        {
            "url": "/"
        }
    ], 
    "paths": {
        "/customers/{customerId}": {
            "post": {
                "parameters": [
                    {
                        "name": "customerId", 
                        "in": "path", 
                        "required": true, 
                        "schema": {
                            "type": "integer"
                        }
                    }
                ], 
                "requestBody": {
                    "content": {
                        "application/json": {
                            "schema": {
                                "type": "object"
                            }
                        }
                    }, 
                    "required": true
                }, 
                "responses": {
                    "200": {
                        "description": "Ok"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /customers/{customerId}:
    post:
      parameters:
      - name: customerId
        in: path
        required: true
        schema:
          type: integer
      requestBody:
        content:
          application/json:
            schema:
              type: object
        required: true
      responses:
        200:
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<h3>JSON</h3>
<pre>
{
    "openapi": "3.0.1", 
    "info": {
        "title": "Swagger Petstore", 
        "version": "1.0.0"
    }, 
    "servers": [
        {
            "url": "/"
        }
    ], 
    "paths": {
        "/customers/{customerId}": {
            "put": {
                "parameters": [
                    {
                        "name": "customerId", 
                        "in": "path", 
                        "required": true, 
                        "schema": {
                            "type": "integer"
                        }
                    }
                ], 
                "requestBody": {
                    "content": {
                        "application/json": {
                            "schema": {
                                "type": "object"
                            }
                        }
                    }, 
                    "required": true
                }, 
                "responses": {
                    "200": {
                        "description": "Ok"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /customers/{customerId}:
    put:
      parameters:
      - name: customerId
        in: path
        required: true
        schema:
          type: integer
      requestBody:
        content:
          application/json:
            schema:
              type: object
        required: true
      responses:
        200:
          description: Ok
</pre>