<h2>Normative - API Definition</h2>
<ul>
  <li><b>NORMA: </b>Definición API - Formatos</li>
  <li><b>NORMA: </b>Definición API - Errores</li>
</ul>
<p>A response not compliant with the standard may cause application issues.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        200:
          description: Ok
        204:
          description: No content
        400:
          description: Not found
        default:
          description: Unknown
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        200:
          description: Ok
          schema:
            $ref: '#/definitions/response'
        204:
          description: No content
        400:
          description: Error
          schema:
            $ref: '#/definitions/response'
        default:
          description: Unknown
definitions:
  response:
    type: object
    allOf:
      - $ref: '#/definitions/standard_response'
    properties:
      data:
        type: object
        properties:
          tipos:
            type: array
            items:
              type: string
    required:
      - data
      - status
  standard_response:
    type: object
    properties:
      status:
        type: object
        properties:
          http_status:
            type: string
          code:
            type: integer
          errors:
            type: array
            items:
              type: object
              properties: 
                name:
                  type: string
                value:
                  type: string
          description:
            type: string
          internal_code:
            type: string
        required:
          - http_status
          - code
          - description  
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.0
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        200:
          description: Ok
        204:
          description: No content
        400:
          description: Not found
        default:
          description: Unknown
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        200:
          description: Ok
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/response'
        204:
          description: No content
        400:
          description: Error
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/response'
        default:
          description: Unknown
components:
  schemas:
    response:
      type: object
      allOf:
        - $ref: '#/components/schemas/standard_response'
      properties:
        data:
          type: object
          properties:
            tipos:
              type: array
              items:
                type: string
      required:
        - data
        - status
    standard_response:
      type: object
      properties:
        status:
          type: object
          properties:
            http_status:
              type: string
            code:
              type: integer
            errors:
              type: array
              items:
                type: object
                properties: 
                  name:
                    type: string
                  value:
                    type: string
            description:
              type: string
            internal_code:
              type: string
          required:
            - http_status
            - code
            - description
            - errors  
</pre>