<p>Depending of the verb, some path structures are not encouraged. This can be a hint of a design flaw.</p>
<h2>Allowed paths</h2>
<table>
  <thead>
    <tr>
      <th>VERB</th>
      <th>STRUCTURE</th>
    </tr>
  </thead>
  <tbody>
  <tr>
    <td>GET</td>
    <td>/resources</td>
  </tr>
  <tr>
    <td>GET</td>
    <td>/resources/{id}</td>
  </tr>
  <tr>
    <td>GET</td>
    <td>/resources/{id}/subresources</td>
  </tr>
  <tr>
    <td>POST</td>
    <td>/resources</td>
  </tr>
  <tr>
    <td>POST</td>
    <td>/resources/get</td>
  </tr>
  <tr>
    <td>POST</td>
    <td>/resources/delete</td>
  </tr>
  <tr>
    <td>POST</td>
    <td>/resources/{id}/subresources</td>
  </tr>
  <tr>
    <td>PUT</td>
    <td>/resources/{id}</td>
  </tr>
  <tr>
    <td>PATCH</td>
    <td>/resources/{id}</td>
  </tr>
  <tr>
    <td>DELETE</td>
    <td>/resources/{id}</td>
  </tr>
  </tbody>
</table>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<h3>YAML</h3>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /resources:
    get:
      responses:
        200:
          description: Ok
    post:
      responses:
        200:
          description: Ok
    put: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR018: Operation not recommended for resource path:  /resources}}</span>
      responses:
        200:
          description: Ok
    patch: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR018: Operation not recommended for resource path:  /resources}}</span>
      responses:
        200:
          description: Ok
    delete: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR018: Operation not recommended for resource path:  /resources}}</span>
      responses:
        200:
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<h3>YAML</h3>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /resources/{r_id}:
    get:
      responses:
        200:
          description: Ok
    put:
      responses:
        200:
          description: Ok
    patch:
      responses:
        200:
          description: Ok
    delete:
      responses:
        200:
          description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<h3>YAML</h3>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /resources:
    get:
      responses:
        200:
          description: Ok
    post:
      responses:
        200:
          description: Ok
    put: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR018: Operation not recommended for resource path:  /resources}}</span>
      responses:
        200:
          description: Ok
    patch: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR018: Operation not recommended for resource path:  /resources}}</span>
      responses:
        200:
          description: Ok
    delete: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR018: Operation not recommended for resource path:  /resources}}</span>
      responses:
        200:
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<h3>YAML</h3>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /resources/{r_id}:
    get:
      responses:
        200:
          description: Ok
    put:
      responses:
        200:
          description: Ok
    patch:
      responses:
        200:
          description: Ok
    delete:
      responses:
        200:
          description: Ok
</pre>
