<p>500 Internal Server Error response codes should not be defined</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
    swagger: "2.0"
    info:
      version: "1.0.0"
      title: "Swagger Petstore"
    host: "example.com"
    basePath: "/"
    schemes:
      - "https"
    consumes:
      - "application/json"
    produces:
      - "application/json"
    paths:
      /status:
        delete:
          summary: "Get API status"
          responses:
            "400":
              description: "error"
      /another:
        delete:
          summary: "Get another API status"
          responses:
            "400":
              description: "error"
      /users:
        delete: 
          summary: "Get all users"
          responses: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR109: 500 Internal Server Error response codes should not be defined}}</span>
            "400":
              description: "Error"
            "500":
              description: "Internal Server Error"
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
    swagger: "2.0"
    info:
      version: "1.0.0"
      title: "Swagger Petstore"
    host: "example.com"
    basePath: "/"
    schemes:
      - "https"
    consumes:
      - "application/json"
    produces:
      - "application/json"
    paths:
      /status:
        delete:
          summary: "Get API status"
          responses:
            "400":
              description: "error"
      /another:
        delete:
          summary: "Get another API status"
          responses:
            "400":
              description: "error"
      /users:
        delete: 
          summary: "Get all users"
          responses: 
            "400":
              description: "Error"
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
    openapi: 3.0.0
    info:
      version: 1.0.0
      title: Swagger Petstore
    paths:
      /status:
        delete:
          summary: Get API status
          responses:
            '400':
              description: error
      /another:
        delete:
          summary: Get another API status
          responses:
            '400':
              description: error
      /users:
        delete:
          summary: Get all users
          responses: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR109: 500 Internal Server Error response codes should not be defined}}</span>
            '400':
              description: Error
            '500':
              description: Internal Server Error
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
    openapi: 3.0.0
    info:
      version: 1.0.0
      title: Swagger Petstore
    paths:
      /status:
        delete:
          summary: Get API status
          responses:
            '400':
              description: error
      /another:
        delete:
          summary: Get another API status
          responses:
            '400':
              description: error
      /users:
        delete:
          summary: Get all users
          responses: 
            '400':
              description: Error
</pre>