<p>Parameters in path shouldn't be numeric</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
    swagger: '2.0'
    info:
      version: 1.0.0
      title: Swagger Petstore
    paths:
      /items/{itemId}:
        get:
          summary: Get item by ID
          parameters:
            - name: itemId
              in: path
              description: ID of the item
              required: false
              type: integer
          responses:
            200:
              description: Item details
              schema:
                type: object
                properties:
                  id:
                    type: integer
                  name:
                    type: string
            400:
                description: error
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
    swagger: '2.0'
    info:
      version: 1.0.0
      title: Swagger Petstore
    paths:
      /items/{itemId}:
        get:
          summary: Get item by ID
          parameters:
            - name: itemId
              in: path
              description: ID of the item
              required: false
              type: string
          responses:
            200:
              description: Item details
              schema:
                type: object
                properties:
                  id:
                    type: integer
                  name:
                    type: string
            400:
              description: error
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
    openapi: "3.0.0"
    info:
      version: 1.0.0
      title: Swagger Petstore
    paths:
      /items/{param1}:
        get:
          summary: Get a list of items
          parameters:
            - name: param1
              in: path
              required: false
              schema:
                type: integer
          responses:
            200:
              description: A list of items
              content:
                application/json:
                  schema:
                    type: array
                    items:
                      type: string
            400:
                description: error
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
    openapi: "3.0.0"
    info:
      version: 1.0.0
      title: Swagger Petstore
    paths:
      /items/{param1}:
        get:
          summary: Get a list of items
          parameters:
            - name: param1
              in: path
              required: false
              schema:
                type: string
          responses:
            200:
              description: A list of items
              content:
                application/json:
                  schema:
                    type: array
                    items:
                      type: string
            400:
                description: error
</pre>