<p>In "get" operations, the $select parameter, which allows to indicate the attributes that will be returned in the response, must be defined as a non-required parameter.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<h3>JSON</h3>
<pre>
{
    "swagger": "2.0", 
    "info": {
        "title": "Swagger Petstore", 
        "version": "1.0.0"
    }, 
    "paths": {
        "/pets": {
            "get": {
                "responses": {
                    "200": {
                        "description": "Ok"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      responses:
        200:
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<h3>JSON</h3>
<pre>
{
    "swagger": "2.0", 
    "info": {
        "title": "Swagger Petstore", 
        "version": "1.0.0"
    }, 
    "paths": {
        "/pets": {
            "get": {
                "parameters": [
                    {
                        "in": "query", 
                        "name": "$select", 
                        "required": false, 
                        "type": "array", 
                        "items": {
                            "type": "string"
                        }
                    }
                ], 
                "responses": {
                    "200": {
                        "description": "Ok"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: query
          name: $select
          required: false
          type: array
          items:
            type: string
      responses:
        200:
          description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<h3>JSON</h3>
<pre>
{
    "openapi": "3.0.1", 
    "info": {
        "title": "Swagger Petstore", 
        "version": "1.0.0"
    }, 
    "paths": {
        "/pets": {
            "get": {
                "responses": {
                    "200": {
                        "description": "Ok"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      responses:
        200:
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<h3>JSON</h3>
<pre>
{
    "openapi": "3.0.1", 
    "info": {
        "title": "Swagger Petstore", 
        "version": "1.0.0"
    }, 
    "paths": {
        "/pets": {
            "get": {
                "parameters": [
                    {
                        "name": "$select", 
                        "in": "query", 
                        "required": false, 
                        "schema": {
                            "type": "array", 
                            "items": {
                                "type": "string"
                            }
                        }
                    }
                ], 
                "responses": {
                    "200": {
                        "description": "Ok"
                    }
                }
            }
        }
    }
}
</pre>
<h3>YAML</h3>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
      - name: $select
        in: query
        required: false
        schema:
          type: array
          items:
            type: string
      responses:
        200:
          description: Ok
</pre>