<p>In "get" operations, the $expand parameter, which allows expanding a sub-element, must be defined as a non-required parameter.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<h3>All the examples are in YAML</h3>
<pre>
  swagger: "2.0"
  info:
    title: Swagger Petstore
    version: 1.0.0
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $expand
            type: array
            required: true
            items:
              type: string
        responses:
          200:
            description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: query
          name: $expand
          type: array
          required: false
          items:
            type: string
      responses:
        200:
          description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
  openapi: 3.0.1
  info:
    title: Swagger Petstore
    version: 1.0.0
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $expand
            required: true
            schema:
              type: array
              items:
                type: string
        responses:
          200:
            description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: query
          name: $expand
          required: false
          schema:
            type: array
            items:
              type: string
      responses:
        200:
          description: Ok
</pre>