<p>The chosen parameter must be defined in this operation. By default, $select</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<h3>All the examples are in YAML</h3>
<pre>
  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters: 
          - in: query
            name: other 
            type: array
            items:
              type: string
          - in: query
            name: hola 
            type: array
            items:
              type: string
        responses:
          206:
            description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $select
            type: array
            items:
              type: string
          - in: query
            name: $select
            type: array
            items:
              type: string
        responses:
          206:
            description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters: 
          - in: query
            name: other 
            schema:
              type: array
              items:
                type: string
          - in: query
            name: select 
            schema:
              type: array
              items:
                type: string
        responses:
          206:
            description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $select
            schema:
              type: array
              items:
                type: string
          - in: query
            name: $select
            schema:
              type: array
              items:
                type: string
        responses:
          206:
            description: Ok
</pre>