<h2>Normative - API Definition</h2>
<ul>
    <li><b>STANDARD: </b>API Definition - Nomenclature</li>
</ul>
<p><b>OpenAPI 2:</b> Base path must be compliant with the standard.</p>
<p><b>OpenAPI 3:</b> The path of 'server' url must be compliant with the standard.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
basePath: /seguros
paths: {}
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
basePath: /api-seguros/v1
paths: {}
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
info:
    version: 1.0.0
    title: Swagger Petstore
servers:
    - url: https://petstore.swagger.io/seguros/version/1
paths: {}
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
info:
    version: 1.0.0
    title: Swagger Petstore
servers:
    - url: https://petstore.swagger.io/api-seguros/v1
paths: {}
</pre>