<p>The convention for naming properties in a JSON is "snake_case" (the words must be in lowercase and separated by an underscore), therefore, the path parameters, the query parameters, request body  properties and response body properties must be in this convention.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<h3>All the examples are in YAML</h3>
<pre>
  swagger: "2.0"
  info:
    title: Swagger Petstore
    version: "1.0"
  paths:
    /pets/{pet-id}:
      patch:
        consumes:
          - application/json
        produces:
          - application/json
        parameters:
          - in: path
            name: pet-id
            type: integer
            required: true
          - in: query
            name: pet_Type
            type: string
          - in: body
            name: pet
            description: Pet
            schema:
              type: object
              required:
                - PET_NAME
              properties:
                PET_NAME:
                  type: string
                photoUrls:
                  type: array
                  items:
                    type: string
        responses:
          200:
            description: OK
            schema:
              type: object
              properties:
                petId:
                  type: integer
                petType:
                  type: string
                petName:
                  type: string
                photoUrls:
                  type: array
                  items:
                    type: string
  </pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
  swagger: "2.0"
  info:
    title: Swagger Petstore
    version: "1.0"
  paths:
    /pets/{pet_id}:
      patch:
        consumes:
          - application/json
        produces:
          - application/json
        parameters:
          - in: path
            name: pet_id
            type: integer
            required: true
          - in: query
            name: pet_type
            type: string
          - in: body
            name: pet
            description: Pet
            schema:
              type: object
              required:
                - pet_name
              properties:
                pet_name:
                  type: string
                photo_urls:
                  type: array
                  items:
                    type: string
        responses:
          200:
            description: OK
            schema:
              type: object
              properties:
                pet_id:
                  type: integer
                pet_type:
                  type: string
                pet_name:
                  type: string
                photo_urls:
                  type: array
                  items:
                    type: string
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets/{pet-id}:
    patch:
      parameters:
      - name: pet-id
        in: path
        required: true
        schema:
          type: integer
      - name: PET_TYPE
        in: query
        schema:
          type: string
      requestBody:
        description: Pet
        content:
          application/json:
            schema:
              required:
              - pet_name
              type: object
              properties:
                pet_name:
                  type: string
                photoUrls:
                  type: array
                  items:
                    type: string
        required: true
      responses:
        200:
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  pet-id:
                    type: integer
                  PET_TYPE:
                    type: string
                  petName:
                    type: string
                  photoUrls:
                    type: array
                    items:
                      type: string
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
  openapi: 3.0.1
  info:
    title: Swagger Petstore
    version: "1.0"
  paths:
    /pets/{pet_id}:
      patch:
        parameters:
        - name: pet_id
          in: path
          required: true
          schema:
            type: integer
        - name: pet_type
          in: query
          schema:
            type: string
        requestBody:
          description: Pet
          content:
            application/json:
              schema:
                required:
                - pet_name
                type: object
                properties:
                  pet_name:
                    type: string
                  photo_urls:
                    type: array
                    items:
                      type: string
          required: true
        responses:
          200:
            description: OK
            content:
              application/json:
                schema:
                  type: object
                  properties:
                    pet_id:
                      type: integer
                    pet_type:
                      type: string
                    pet_name:
                      type: string
                    photo_urls:
                      type: array
                      items:
                        type: string
</pre>