<p>All parameters in query must be snake_case</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /invoices:
      get:
        responses:
          200:
            description: A invoice.
            schema:
              type: object
              properties:
                product:  
                  type: string 
                  format: int128
                line: 
                  type: string 
                lapiz:
                  type: string
                price:  
                  type: string
                  format: byte
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /invoices:
      get:
        responses:
          200:
            description: A invoice.
            schema:
              type: object
              properties:
                product:  
                  type: string 
                  format: byte
                line: 
                  type: string 
                  format: byte
                lapiz:
                  type: string
                  format: byte
                price:  
                  type: string
                  format: byte
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
  openapi: "3.0.0"
  info:
    version: "1.0.0"
    title: "Swagger Petstore"
  paths:
    /invoices:
      get:
        responses:
          '200':
            description: A invoice.
            content: 
              application/json:
                schema:
                  type: object
                  properties:
                    product:  
                      type: string 
                      format: int128
                    line: 
                      type: string 
                    price:
                      type: string
                      format: byte
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
  openapi: "3.0.0"
  info:
    version: "1.0.0"
    title: "Swagger Petstore"
  paths:
    /invoices:
      get:
        responses:
          '200':
            description: A invoice.
            content: 
              application/json:
                schema:
                  type: object
                  properties:
                    product:  
                    type: string
                    format: byte
                    line: 
                      type: string
                      format: byte 
                    price:
                      type: string
                      format: byte
</pre>