<p>When defining an API, the response data formats must be defined.</p>
<p>According to the standard, the default format will be application/json, with some exceptions.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
produces:
  - application/xml
paths:
  /pets:
    put:
      responses:
        200:
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
produces:
  - application/json
paths:
  /pets:
    put:
      responses:
        200:
          description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses:
        200:
          description: Ok
          content:
            application/xml:
              schema:
                type: object
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses:
        200:
          description: Ok
          content:
            application/json:
              schema:
                type: object
</pre>