<p>All summaries must begin with a capital letter and end with a period</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
  swagger: '2.0'
  info:
    title: Sample API
    version: 1.0.0
    description: Esta es la descripción de la API.
  paths:
    /pets:
      get:
        summary: Lista de mascotas <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR087: All summaries must begin with a capital letter and end with a period}}</span>
        description: Esta ruta devuelve una lista de mascotas.
        responses:
          '200':
            description: Una lista de mascotas retornada con éxito.
            schema:
              $ref: '#/definitions/Pet'
  definitions:
    Pet:
      type: object
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
      description: Representa una mascota.
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
  swagger: '2.0'
  info:
    title: Sample API
    version: 1.0.0
    description: Esta es la descripción de la API.
  paths:
    /pets:
      get:
        summary: Lista de mascotas.
        description: Esta ruta devuelve una lista de mascotas.
        responses:
          '200':
            description: Una lista de mascotas retornada con éxito.
            schema:
              $ref: '#/definitions/Pet'
  definitions:
    Pet:
      type: object
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
      description: Representa una mascota.
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
  openapi: 3.0.0
  info:
    title: Sample API
    version: 1.0.0
    description: Esta es la descripción de la API.
    summary: Este es un resumen del API <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR087: All summaries must begin with a capital letter and end with a period}}</span>
  
  paths:
    /pets:
      get:
        summary: lista de mascotas. <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR087: All summaries must begin with a capital letter and end with a period}}</span>
        description: Esta ruta devuelve una lista de mascotas.
        responses:
          '200':
            description: Una lista de mascotas retornada con éxito.
  
  components:
    schemas:
      Pet:
        type: object
        properties:
          id:
            type: integer
            format: int64
          name:
            type: string
        description: Representa una mascota.
        summary: Resumen de una mascota <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR087: All summaries must begin with a capital letter and end with a period}}</span>
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
  openapi: 3.0.0
  info:
    title: Sample API
    version: 1.0.0
    description: Esta es la descripción de la API.
    summary: Este es un resumen del API.
  
  paths:
    /pets:
      get:
        summary: Lista de mascotas.
        description: Esta ruta devuelve una lista de mascotas.
        responses:
          '200':
            description: Una lista de mascotas retornada con éxito.
  
  components:
    schemas:
      Pet:
        type: object
        properties:
          id:
            type: integer
            format: int64
          name:
            type: string
        description: Representa una mascota.
        summary: Resumen de una mascota.
</pre> 